flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -100,6 +116,35 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
       });
     }
   }
+
+  void _startTransaction(String? name, Object? arguments) {
+    if (!_enableAutoTransactions) {
+      return;
+    }
+    if (name == null) {
+      return;
+    }
+    _hub.configureScope((scope) {
+      if (scope.span == null) {
+        _transaction = _hub.startTransaction(
+          name,
+          'ui.load',
+          bindToScope: true,
+          waitForChildren: true,
+        );
+        if (arguments != null) {
+          _transaction?.setData('route_settings_arguments', arguments);
+        }
+        _transaction?.finishAfter(Duration(seconds: 3),
+            status: SpanStatus.ok());
+      }
```

Comment:
```suggestion
      _transaction = _hub.startTransaction(
        name,
        'ui.load',
        bindToScope: scope.span == null,
        waitForChildren: true,
      );
      if (arguments != null) {
        _transaction?.setData('route_settings_arguments', arguments);
      }
      _transaction?.finishAfter(Duration(seconds: 3),
          status: SpanStatus.ok());
```

If the scope already has a span you still should create a transaction. 
Just don't replace it on the scope.

Reply:
See https://github.com/getsentry/sentry-dart/pull/643#discussion_r761092226